app/views/NewExploreDeep/ProjectContent/index.tsx

```diff
@@ -0,0 +1,291 @@
+
+import styles from './styles.css';
+
+interface ResolutionOption {
+    key: 'day' | 'month' | 'year';
+    label: React.ReactNode;
+}
+
+const resolutionKeySelector = (d: ResolutionOption) => d.key;
+const resolutionLabelSelector = (d: ResolutionOption) => d.label;
+
+const resolutionOptions: ResolutionOption[] = [
+    {
+        key: 'day',
+        label: 'D',
+    },
+    {
+        key: 'month',
+        label: 'M',
+    },
+    {
+        key: 'year',
+        label: 'Y',
+    },
+];
+
+interface ChartTypeOption {
+    key: 'step' | 'spark';
+    label: React.ReactNode;
+}
+const chartTypeKeySelector = (d: ChartTypeOption) => d.key;
+const chartTypeLabelSelector = (d: ChartTypeOption) => d.label;
+
+const chartTypeOptions: ChartTypeOption[] = [
+    {
+        key: 'step',
+        label: <IoStatsChartSharp />,
+    },
+    {
+        key: 'spark',
+        label: <IoTrendingUpSharp />,
+    },
+];
+
+type Timeseries = {
+    date: string;
+    projectCount: number;
+}
+
+type TimeseriesItem = {
+    date: number;
+    total: number;
+}
+
+function timeSpentLabelFormatter(value: number) {
+    return [value, 'Total projects'];
+}
+
+interface Props {
+    className?: string;
+    timeseries: Timeseries[] | undefined;
+}
+
+function ProjectContent(props: Props) {
+    const {
+        className,
+        timeseries,
+    } = props;
+
+    const [activeView, setActiveView] = useState<'map' | 'table' | undefined>('map');
+    const [resolution, setResolution] = React.useState<'year' | 'month' | 'day'>('day');
+    const [chartType, setChartType] = React.useState<'step' | 'spark'>('spark');
+
+    const timeseriesData = useMemo(
+        () => {
+            const values = (timeseries ?? [])
+                .filter((item) => isDefined(item.date))
+                .map((item) => ({
+                    date: resolveTime(item.date, resolution).getTime(),
+                    total: item.projectCount,
+                }))
+                .filter((item) => item.total > 0);
+
+            return mergeItems(
+                values,
+                (item) => String(item.date),
+                (foo, bar) => ({
+                    date: foo.date,
+                    total: foo.total + bar.total,
+                }),
+            ).sort((a, b) => (a.date - b.date));
+        },
+        [timeseries, resolution],
+    );
+
+    // eslint-disable-next-line no-nested-ternary
+    const timeFormatter = resolution === 'day'
+        ? formatDate
+        : (resolution === 'month' ? formatMonth : formatYear);
+
+    const timeseriesWithoutGaps: TimeseriesItem[] | undefined = useMemo(
+        () => {
+            if (!timeseriesData) {
+                return undefined;
+            }
+
+            if (timeseriesData.length <= 0) {
+                return [
+                    {
+                        total: 0,
+                        date: resolveTime(new Date(), resolution).getTime(),
+                    },
+                ];
+            }
```

Comment:
Let's see if both of these conditions are the same and should be handled together